apps/emqx_connector/src/emqx_connector_mongo.erl

```diff
@@ -256,6 +258,14 @@ on_query(
             {ok, Result}
     end.
 
+%% For some reason the collection name is turned into an atom by the emqx_authz
+%% application which causes the mongo query to break. TODO: This should probably
+%% be fixed at another level.
+make_collection_bin(Atom) when is_atom(Atom) ->
```

Comment:
the field is of type `binary()` if it ended up as an atom, we should investigate (because it could be a potential atom leak here).

Reply:
I will try to investigate this.

Reply:
Fixed by changing the authz schema (see new commit below). I don't understand why the authz schema don't piggyback on the mongodb connector schema but maybe there is a good reason for that?